### PR TITLE
Fix ready event bug

### PR DESCRIPTION
**Pull request by ryardley**
Closes: #614 

### What are the changes and their implications?
* Ready event was firing too early in certain situations when chokidar and vfs globbing was too slow. (symlinks/large gitignore/low memory/low cpu)
* Refactors the way the ready event works. The ready event is fired when the agnosticinput stream switches from using the vinyl-fs glob to the chokidar watcher instead of after a arbitrary length of pause time. 
* Adds universal transform helper to more stages
* This is still not perfect until we add ordering to the queue so that the event only fires after the previous event has actually been written (the nature of streams means events can be processed out of order). We can do that in a later PR as this gets us close enough for most cases because of the way webpack and next work.

### Checklist

- [x] Tests added for changes
- [x] User facing changes documented

### CODE REVIEW

File: packages/file-pipeline/src/helpers/agnostic-source/agnostic-source.test.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      if (data === 'ready') {
+        stream.end()
+        expect(log).toEqual(expected)
+        return done()
+      }
+      log.push(data.path)
+    })
+  })
+
+  test.skip('when watching = true', async () => {
```
Comment (Skn0tt) [inline packages/file-pipeline/src/helpers/agnostic-source/agnostic-source.test.ts:36]: Is there a specific reason this test is skipped?

Comment (ryardley) [reply packages/file-pipeline/src/helpers/agnostic-source/agnostic-source.test.ts:36]: Nice catch!

**Comment by ryardley** [comment]
So in order to run a test that actually uses chokidar (despite closing the watcher properly) it seems we have to increase the available file system watchers in linux. So I have done that here. Please review again. If you are happy I can merge. @flybayer @merelinguist @Skn0tt

**Comment by Skn0tt** [comment]
LGTM, provided that we investigate number of watchers in the future.

**Pull request merged**